### ISSUE TITLE
add_dc_field -- APIError:  HTTP Status Code 400

### ISSUE THREAD

**Issue by kurt-rhee** [stephenkaplan/plantpredict-python#103]
Hey Stephen, 

Thanks again for making this API!  It is going to seriously speed up our workflow. 

I'm writing my first script with the API and I get the following when assigning a dc field to the powerplant object.  Feel free to ignore my commented messages within dc_fields, they are just reminders for me to check which units to use in the online UI.

```
APIError: HTTP Status Code 400: b'{"message":"The request is invalid.","modelState":{"post Height":["Post Height is required."]}}'
```

```
Version 0.9.2

standard_dc_field = powerplant.add_dc_field(
        block_name = standard_block,
        array_name = standard_array,
        inverter_name = standard_inverter,
        module_id = module_id,
        ground_coverage_ratio = gcr,
        number_of_series_strings_wired_in_parallel = strings_in_parallel,
        field_dc_power = 1000, # dc power must be calculated,
        tracking_type = TrackingTypeEnum.HORIZONTAL_TRACKER,
        dc_field_backtracking_type = tracking_algorithm['module'],
        modules_high = 1,
        modules_wired_in_series = mods_per_string,
        number_of_rows = 100,
        module_tilt = 0,
        minimum_tracking_limit_angle_d = -60,
        maximum_tracking_limit_angle_d = 60,
        module_orientation = 1, # what does this mean?
        lateral_intermodule_gap = 0.02, # what units is this
        vertical_intermodule_gap = 0.02, # what units is this
#        planned_module_rating = planned_module_rating,
        array_based_shading = True, # what does this mean
        module_quality = 1.3, # change this in the online UI definition
        tracker_load_loss = tracker_load_loss,
        dc_wiring_loss_at_stc = dc_wiring_loss_at_stc,
        dc_health = dc_health
        )
```

**Comment by stephenkaplan** [comment]
@kurt-rhee happy to hear you have been using it. I am more than happy to help you debug this. 2 questions:

1. Looks like you have version 0.9.2...but just to confirm, did you install via ```pip``` or directly from Github? And how recently?
2. Is this the code actually triggering the error? or does the error occur when you follow this with a call `powerplant.update()``` 

I am happy to help you sort this bug out with the current version you are using, but just an FYI - I am currently working on version ```0.9.3``` that is dedicated to improving the "power plant builder" functions (```add_dc_field```, ```add_inverter```, etc.). The next version will eliminate bugs, improve documentation, and also slightly change the usage of these functions (which might break your code, but I can assist with that, and it will overall be more straightforward to use). 

Also, I know you said to ignore the comments, but if you'd like me to clear any of those questions up, please let me know. Once again, some of those questions will hopefully be cleared up in the documentation improvements I'm working on.

**Comment by kurt-rhee** [comment]
Yeah!  I promised we would use it once it was python 3 compatible, so I'm keeping my promise.

1. I installed via pip into a conda environment
``` plantpredict              0.9.2                    pypi_0    pypi```

2. The code snippet itself doesn't cause an error, the error occurs once I use:
```powerplant.create()```

3. It is easy enough to have a code window open as well as the plantpredict UI and cross reference for units, but just having one window would be nice.

**Comment by stephenkaplan** [comment]
@kurt-rhee are you using a bifacial module in this DC field? I wrote this part of the SDK before bifacial modules were implemented in PlantPredict, which is why ```post_height``` isn't a required input to ```add_dc_field()```. However, in PlantPredict you must specify the following dc field parameters when you use a bifacial module:

- ```post_height``` [units meters]
- ```structure_shading``` [units % loss]
- ```backside_mismatch``` [units % loss]

With the current version of the SDK, you will have to add them after the fact. Here's an example of how to do that, assuming you are working with the first block in the list of blocks, first array, first inverter, first dc field (an index of 0 in this case refers to a block with the name "1", etc.)

```python

powerplant.blocks[0]['arrays'][0]['inverters'][0]['dc_fields'][0]['post_height'] = # structure post height
powerplant.blocks[0]['arrays'][0]['inverters'][0]['dc_fields'][0]['backside_mismatch'] = # enter loss
powerplant.blocks[0]['arrays'][0]['inverters'][0]['dc_fields'][0]['structure_shading'] = # enter loss

```

As I mentioned, the next version of the SDK is focused on power plant builder improvements, so I will file this as a bug/improvement and get something in there.
